File: app/src/main/java/com/alpha/acamera/camera/CameraControl.java
package com.alpha.acamera.camera;

import android.view.View;

import com.alpha.acamera.camera.widget.ResizeAbleSurfaceView;

public abstract class CameraControl {
    public abstract void openCamera(int cameraId);

    public abstract void startPreview(ResizeAbleSurfaceView surfaceView);

    public abstract boolean isOpen();

    public abstract boolean isOpening();

    public abstract void takePicture(PictureCallback callback);

    public abstract void closeCamera();

    public abstract void setErrListener(ErrorListener mErrListener);

    public abstract void onPreviewFrame(PreviewCallback callback);
}

File: app/src/main/java/com/alpha/acamera/camera/CameraControl.kt
package com.alpha.acamera.camera

import com.example.android.camera.utils.AutoFitSurfaceView

interface CameraControl {
    fun openCamera(cameraId: Int)
    fun startPreview(surfaceView: AutoFitSurfaceView?)
    val isOpen: Boolean
    val isOpening: Boolean
    fun takePicture(callback: PictureCallback?)
    fun closeCamera()
    fun setErrListener(mErrListener: ErrorListener?)
    fun onPreviewFrame(callback: PreviewCallback?)
}